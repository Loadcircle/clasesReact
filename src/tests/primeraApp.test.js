const { render } = require("@testing-library/react");
import { shallow } from 'enzyme';
import React from 'react';
const { default: PrimeraApp } = require("../PrimeraApp");
// import '@testing-library/jest-dom'; necesario para el snatshop

describe('Pruebas en primera app',()=>{

    test('debe mostrar el mensaej "Hola, Soy Goku"', ()=>{
       
        const mensaje = "Hola, Soy Goku";

        // Esta es una forma 
        // const wrapper = render( <PrimeraApp saludo={mensaje} /> )
        // wrapper.getByText()
        
        //Tambi√©n se puede destructurar

        const {getByText} = render( <PrimeraApp saludo={mensaje} /> )

        //para utilizar el toBeInTheDocument e tiene que expandir el document
        expect(getByText(mensaje)).toBeInTheDocument();

    });

    
    test('Debe de mostrar primeraApp correctamente', ()=>{

        const mensaje = "Hola, Soy Goku";
        const wrapper = shallow(<PrimeraApp saludo={mensaje} />);

        expect(wrapper).toMatchSnapshot()

    });


    test('debe de mostrar el primer parrafo del componente', ()=>{
    
        const mensaje = "Hola, Soy Goku";
        // const subtitulo = "Este es un subtitulo";
        const wrapper = shallow(
            <PrimeraApp
                saludo={mensaje} 
            />);

        const textoParrafo = wrapper.find('p').first().text();

        console.log(textoParrafo)

        expect(textoParrafo).toBe('123456')

    });

});