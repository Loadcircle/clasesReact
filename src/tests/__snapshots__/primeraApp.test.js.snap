// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en primera app Debe de mostrar primeraApp correctamente 1`] = `
<Fragment>
  <h1>
    Hola, Soy Goku
  </h1>
  <p>
    1
    2
    3
    4
    5
    6
  </p>
  <p>
    Jesus
  </p>
</Fragment>
`;
